frontend/src/pages/RoomPage.tsx

```diff
@@ -1,30 +1,61 @@
-import { useEffect, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 import { useNavigate, useParams } from "react-router-dom";
-import { Avatar, Divider, Grid, Stack, Typography } from "@mui/material";
+import { Avatar, Divider, Grid, Stack } from "@mui/material";
 import { useSnackbar } from "notistack";
 
+import { SOCKET_IO_DISCONNECT_REASION } from "src/constants/socket.io";
+import { useAuth } from "src/contexts/AuthContext";
 import { useSocket } from "src/contexts/WsContext";
+import { useGetUserName } from "src/hooks/useUsers";
+import { nameToInitials } from "src/utils/string";
 
 import { Editor } from "../components/Editor";
 import { Question } from "../components/Question";
 import { StyledButton } from "../components/StyledButton";
 
 import { ROOM_EVENTS, ROOM_NAMESPACE } from "~shared/constants";
+import {
+  JoinedPayload,
+  PartnerDisconnectPayload,
+  PartnerLeavePayload,
+} from "~shared/types/api";
+
+type participant = {
+  userId: number;
+  name?: string;
+  isConnected: boolean;
+};
 
 export const RoomPage = () => {
   const { roomId } = useParams();
+  const { user } = useAuth();
   const { socket, connect } = useSocket();
-  const [leaveRoom, setIsLeaveRoom] = useState<boolean>(false);
-  const [message, setMessage] = useState<string>("");
+  const [self, setSelf] = useState<participant>({
+    // We know that if the page renders, user is not null.
+    userId: user?.userId || NaN,
+    name: user?.name || "",
+    isConnected: false,
+  });
+  const [partner, setPartner] = useState<participant | undefined>(undefined);
```

Comment:
Should we standardize whether to use undefined or null? Wen Ling's PR uses null in the SelectDifficulty page